app/Meta.py

```diff
@@ -0,0 +1,564 @@
+        columns=data.columns.values.tolist()
+        accepted_columns=['name','start_date','end_date','description','design']
+     
+        if(all(x in accepted_columns for x in columns)):
+            data['user_id']=user_id
+            data['design'].fillna('some text', inplace=True)
+            insert_table(table='experiments',data=data)
+            msg = {'Message' : 'Successfully inserted',
+                   'Table Affected' : 'Experiments',
+                    'Lines Inserted': data.shape[0]}
+            
+            return make_response(jsonify(msg), 201)
+
+        else:
+            msg = {'Message' : 'File not acceptable.Check the format of file or columns'}
+            return make_response(jsonify(msg), 400)
+    
+    except OperationalError:
+        # Logs the error appropriately
+        logging.error(traceback.format_exc())
+        return 500
+    except IntegrityError:
+        # Logs the error appropriately
+        logging.error(traceback.format_exc())
+        return 409
+    except Exception as e:
+        # Logs the error appropriately
+        logging.error(traceback.format_exc())
+        return 410     
+     
+def insert_sites(fileName,shp_file,dbf_file,prj_file,shx_file):
+    """
+    This function responds to a request for  /yaba/v1/sites with csv and shape files.Checks
+    file is appropriate one.Takes all the shape file.Extracts geometry from Geopandas dataframe
+    and then added to Pandas dataframe.Insertion is done to "sites" table
+
+
+    :shp_file:      .shp file
+    :dbf_file:      .dbf file
+    :prj_file:      .prj file
+    :shx_file:      .shx file
+    :fileName:      CSV file with Sites meta data
+    :return:        201 on success
+                    400 if file is unsuitable or does not contain appropriate columns
+                    409 Intregrity or Constraint error : 23503 foreign_key_violation | 23505 unique_violation
+                    500 Database Connection Error
+                    401 Unauthorized | No user exists
+                    410 Default error.See logs for more information
+    """
+    try:
+        save_tempFile(shp_file)
+        save_tempFile(dbf_file)
+        save_tempFile(prj_file)
+        save_tempFile(shx_file)
+
+        #Getting the shp file from temp folder
+        shp_file_target = os.path.join(os.getcwd(),UPLOAD_FOLDER,shp_file.filename)
+    
+        #Reading the shapefile as DataFrame
+        data_g=gpd.read_file(shp_file_target)
+    
+        #Reading the csv as Dataframe
+        data = pd.read_csv(fileName,delimiter = ',')
+
+        
+        data['geometry'] = data.geometry.astype(str)
+        
+        geos.WKBWriter.defaults['include_srid'] = True
+
+        
+        for i in range(data_g.shape[0]):
+            p=data_g.iat[i,10]
+            geos.lgeos.GEOSSetSRID(p._geom, 4326)
+            data.iat[i,6]=p.wkb_hex
+
+        #Checking necessary columns are there.
+        columns=data.columns.values.tolist()
```

Comment:
I suggest moving this check to immediately after the csv file is loaded (removing the 'geometry' column that's added later) and immediately returning if the check fails.